GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -523,7 +523,32 @@ public bool TryLoad(out string error)
                 this.UpgradeRing = RingType.NoConfig;
 
                 string ringConfig = null;
-                if (this.LocalConfig.TryGetConfig(GVFSConstants.LocalGVFSConfig.UpgradeRing, out ringConfig, out error))
+                string upgradeFeedUrl = null;
+                bool feedURLConfigured = false;
+
+                // There are couple config settings that contain info about Upgrade rings.
+                // upgrade.ring and upgrade.feedURL. upgrade.feedURL will override upgrade.ring
+                // config. If upgrade.feedURL is set, but does not contain valid ring info then
+                // upgrade ring would be set to NoConfig and auto upgrade checks will fail.
+                if (this.LocalConfig.TryGetConfig(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl, out upgradeFeedUrl, out error) &&

```

Comment:
If a feed is configured and it does not contain a valid ring this method will return `false` with the error "error = "Could not read GVFS Config."

Is that expected (we were able to read the GVFS config)?

Reply:
good catch. we can provide more granular error messaging.

Reply:
Is it expected that this method will return false in that scenario?  I think some of my confusion is coming from this method being used for two different purposes (determining the ring and reading the config when a user is configured to use the GitHub upgrader).

Reply:
Actually, based on the unit tests it looks like `false` is expected?  I am not sure I understand when this method will return `true` vs `false`

Reply:
On further digging, it turns out `TryLoad` returns `false` only when there is a file i/o error while attempting to read `gvfs.config`. Otherwise it should return `true`. I restructured the code a little to make this more obvious. Also removed the new code where I was incorrectly returning `false` on `feedurl` parse errors. Fixed couple UT as well where I was wrongly checking the `TryLoad` return code.